python/tvm/topi/arm_cpu/depthwise_conv2d.py

```diff
@@ -394,7 +394,8 @@ def schedule_conv_out(out):
             ci_outer, ci_inner = s[out].split(ci, 4)
             s[out].vectorize(ci_inner)
             s[out].unroll(ci_outer)
-
+        else:
+            s[out].vectorize(ci)
```

Comment:
Shouldn't be the value of `ci` limited somehow? What if the value of `ci` is huge? You can use `filter` argument in `define_split` to prevent this situation.

Reply:
This is a good point. I'm not sure what would happen if `ci` was huge, I guess LLVM would legalize the vectors and loop over the size of the vector registers in hardware when they are storing 32bit floats.

I sort of just copied this from [above](https://github.com/apache/tvm/blob/main/python/tvm/topi/arm_cpu/depthwise_conv2d.py#L369) though using the same scheduling that we use for the actual convolution.

Since the fallback scheduling is to use a loop splitting where `ci = 8` and for the hardware in question this is the number of 32bit float in a vector I think we should be okay. In the case we aren't using the fallback i.e. auto-scheduling I would hope the auto-scheduler is able to find the optimal choice for `ci` which should be at least as good as `8` or better, but perhaps there are scenarios where it could choose a bad value?

We could do something similar to how we handle regular convolutions, and define a [tunable parameter for vectorization](https://github.com/apache/tvm/blob/main/python/tvm/topi/arm_cpu/conv2d_spatial_pack.py#L324), but that would potentially change the scheduling for the depthwise convolution here, rather than its output, which is sort of beyond the initial scope of this PR.

What are your thoughts?

Reply:
Thank you for your answer. I think that to avoid guesses about the size of `ci` (are use sure that it will be equal to `8`?) you can just add a filter parameter to define split, e.g.: 
```python
cfg.define_split("tile_c", c, num_outputs=2, filter=lambda entry: entry.size[1] <= 32)
```
In this case I believe we can guarantee that `ci` won't be bigger than 32. What do you think?

Reply:
I think we are already defining a split [here](https://github.com/apache/tvm/blob/main/python/tvm/topi/arm_cpu/depthwise_conv2d.py#L311) though, this is where I got the value of `8` from.

Reply:
This split will work only for run w/o tuning statistic (in default mode). I suggest restricting the search space in tuning by adding filter value in `define_split`. It will help to improve tuning time and dropout useless configurations. 
By the way, I reread your message and noticed that probably I was wrong than suggest to restrict vectorization size by 32. You wrote that it is storing 32bit floats, that means that the maximum possible capacity for vectorization is 8, am I right?

Reply:
Cool, thanks for the suggestion, I've added a filter on the split. 

Yeah I think you are almost correct, for neon where vectors are `128` bits and `sizeof(float) = 32`, then there will be `128 / 32 = 4` elements processed in each vector operation. I think the backend will split the 8 element llvm vectors into two 4 element vectors during legalization, so this isn't a problem.